internal/types/document.go

```diff
@@ -277,6 +278,40 @@ func (d *Document) Remove(key string) {
 	panic(fmt.Sprintf("types.Document.Remove: key not found: %q", key))
 }
 
+// RemoveByPath removes document by path, doing nothing if the key does not exist.
+func RemoveByPath(d *Document, keys ...string) {
+	if len(keys) == 0 {
+		return
+	}
+
+	key := keys[0]
+	if _, ok := d.m[key]; !ok {
+		return
+	}
+
+	if len(keys) == 1 {
+		d.Remove(key)
+		return
+	}
+
+	key2 := keys[1]
+	switch x := d.m[key].(type) {
+	case *Document:
+		delete(x.m, key2)
```

Comment:
That's not enough: we should remove from x.keys. Overall, let's use existing methods: Get, Remove, etc. They have all checks

Reply:
fixed.